packages/next/src/client/components/http-error-fallback-boundary.tsx

```diff
@@ -1,36 +1,54 @@
 'use client'
 
+/**
+ * HTTPErrorFallbackBoundary is a boundary that catches errors and renders a
+ * fallback component for HTTP errors.
+ *
+ * It receives the status code, and determine if it should render fallbacks for few HTTP 4xx errors.
+ *
+ * e.g. 404
+ * 404 represents not found, and the fallback component pair contains the component and its styles.
+ *
+ * TODO: support 401 and 403 HTTP errors.
+ */
+
 import React, { useContext } from 'react'
 import { useUntrackedPathname } from './navigation-untracked'
 import { isNotFoundError } from './not-found'
 import { warnOnce } from '../../shared/lib/utils/warn-once'
 import { MissingSlotContext } from '../../shared/lib/app-router-context.shared-runtime'
 
-interface NotFoundBoundaryProps {
+const HTTPErrorStatus = {
+  NOT_FOUND: 404,
+  // TODO: support 401 and 403 HTTP errors.
+} as const
+
+interface HTTPErrorFallbackBoundaryProps {
   notFound?: React.ReactNode
-  notFoundStyles?: React.ReactNode
+  errorStatus?: number
```

Comment:
That prop is unused, isn't it? I also don't understand how it _could_ be used, since the status code is controlled by the error that is caught by this boundary.

Reply:
yea it was the legacy as the removed `asNotFound`, missed removing them here